md

```diff
@@ -0,0 +1,155 @@
+# New API for conversion from UTF-8 to UV
+
+## Preamble
+
+    Author:  K. H. Williamson <khw@cpan.org>
+    ID:      RFC-0022
+    Status:  Draft
+
+
+## Abstract
+
+Introduce an API more convenient to use safely
+
+## Motivation
+
+The existing API requires disambiguation between a NUL character and malformed
+UTF-8, which callers tend to not do, and the caller besides has to take
+extra steps to correctly implement current best practices for dealing with
+malformed UTF-8.
+
+## Rationale
+
+This API has no ambiguity between success and failure, and always returns 
+based on best practices. Other proposals were discarded in the pre-RFC process:
+[Pre-RFC: New C API for converting from UTF-8 to code point](http://nntp.perl.org/group/perl.perl5.porters/264207)
+
+## Specification
+
+I think it best to start with just the two functions that will be applicable in
+almost all situations.  The pod for these is
+
+=for apidoc      |bool|utf8_to_uvchr       |const char * s|const char * e|UV *cp|Size_t * len
+=for apidoc_item |bool|utf8_to_uvchr_nowarn|const char * s|const char * e|UV *cp|Size_t * len
+
+These each translate UTF-8 into UTF-32 (or UTF-64 on platforms where a UV is 64
+bits long), returning <true> if the operation succeeded without problems; and
+false otherwise.  (Also, they operate on UTF-EBCDIC rather than UTF-8 on
+EBCIDIC platforms.)
+
+They differ in how they handle input which results in a <false> return.
+
+More precisely, they each calculate the first code point represented by the
+sequence of bytes bounded by <*s> .. <(*e) - 1>, interpreted as UTF-8.
+<e> must be strictly greather than <s>; this is asserted for in debugging
+builds.
+
+Since UTF-8 is a variable length encoding, the number of bytes examined also
+varies.  The algorithm is to first look at <*s>.  If that represents a full
+UTF-8 character, no other byte is examined, and the UTF-8 is calculated from
```

Comment:
"and the UTF-8 is calculated"

should this be:

"and the code point is calculated"
?

Reply:
Yes, thanks